lib/core/connector_settings.rb

```diff
@@ -64,6 +64,22 @@ def [](property_name)
       @elasticsearch_response[:_source][property_name]
     end
 
+    def features
+      self[:features]
+    end
+
+    # being defensive here to avoid regressions, will be subject to change in the near future
+    def filtering_rule_feature_enabled?
+      features && !features.empty? && features[:filtering_rules]
+    end
+    def filtering_advanced_config_feature_enabled?
+      features && !features.empty? && features[:filtering_advanced_config]
```

Comment:
```suggestion
      features[:filtering_advanced_config]
```

or 

```suggestion
      !!features[:filtering_advanced_config]
```

Reply:
Both suggestions break the tests when features were tempered with. That was the reason why I was so defensive here and checked for multiple cases.